회원가입 - 로그인

Jeonminje opened the pull request:
로그인 및 로그아웃을 세션을 이용해 구현

Jeonminje committed:
SignUp-login

로그인 및 로그아웃을 세션을 이용해 구현

sunwootest commented on movieservice/src/main/java/study/movieservice/controller/LoginController.java:
```diff
@@ -0,0 +1,42 @@
+        if(session != null){
+            session.invalidate();
+        }
+        return "로그아웃 되었습니다.";
```
Presentation Layer에 비즈니스 로직이 담겨져 있는 것 같네요 :)

Jeonminje replied:
Layered Architecture 틀을 유지하도록 노력하겠습니다!

sunwootest commented on movieservice/src/main/java/study/movieservice/service/MemberService.java:
```diff
@@ -40,4 +41,24 @@ public boolean checkLoginId(String loginId){
         return false;
     }
 
+    public boolean checkMember(String loginId, String password, HttpServletRequest request){
+
```
checkMember라는 메소드를 봤을 때 어떤 기능을 하는지 딱 떠오를까요? 메소드 명은 부연설명없이 이름만으로 기능을 알 수 있어야 합니다.

sunwootest replied:
Member와 관련된 비즈니스 로직을 처리해야할 메소드인데 세션과 서블릿에 의존적인 메소드가 되어버렸네요. 어떤 방법으로 메소드를 작성해야 유지보수가 용이할지 고민하면 좋겠습니다.

Jeonminje replied:
좀더 고민하도록 하겠습니다

sunwootest commented on movieservice/src/main/java/study/movieservice/controller/LoginController.java:
```diff
@@ -0,0 +1,42 @@
+public class LoginController {
+
+    private final MemberService memberService;
+
+    @ResponseBody
```
@ResponseBody를 매번 반복할 필요가 없어보이는데 이전 PR에서 알려드렸던 내용으로 개선해주세요 :)

Jeonminje replied:
넵 감사합니다

sunwootest reviewed (commented):
세션을 이용해서 로그인을 잘 구현하셨습니다. Layered Architecture의 틀을 유지하며 유지보수가 용이한 코드로 잘 작성해봅시다 :)

Jeonminje committed:
Merge branch 'SignUp-emailCheck-minje' into SignUp-login-minje

# Conflicts:
#	movieservice/src/main/java/study/movieservice/service/MemberService.java
#	src/main/java/study/movieservice/controller/LoginController.java
#	src/main/java/study/movieservice/domain/SessionConst.java

Jeonminje committed:
로그인 기능 수정 및 이메일 중복확인 기능과 merge

## LoginController
- 로그인과 로그아웃 요청이 들어오면 MemberService의 함수로 처리하였습니다.

## MemberMapper
- getMember() 함수를 추가하였습니다.

## MemberService
- logIn() logOut() 함수를 추가해 세션을 이용한 로그인, 로그아웃을 구현하였습니다.

Jeonminje commented:
커밋하였습니다!

Jeonminje commented:
## LoginController
- 로그인과 로그아웃 요청이 들어오면 MemberService의 함수로 처리하였습니다.

## MemberMapper
- getMember() 함수를 추가해 loginId 로 DB에서 member 객체를 리턴해 주었습니다. 지금은 단순 로그인을 위한 용도로 사용하지만 추후에 다른용도로도 재사용할 경우를 대비해 member 객체를 리턴 했습니다.

## MemberService
- logIn() logOut() 함수를 추가해 세션을 이용한 로그인, 로그아웃을 구현하였습니다.

sunwootest commented on src/main/java/study/movieservice/service/MemberService.java:
```diff
@@ -44,4 +47,28 @@ public void sendSignUpMessage(String email){
         mailService.sendMail(message);
     }
+
+    public boolean logIn(String loginId, String password, HttpServletRequest request){
+
+        Optional<Member> finding  = memberMapper.getMember(loginId);
+
+        if(finding.isPresent()){
+            Member findMember = finding.get();
+            if(BCrypt.checkpw(password, findMember.getLoginPassword())){
+
+                HttpSession session = request.getSession();
+                session.setAttribute(findMember.getLoginId(), findMember.getNickname());
+
+                return true;
+            }
+        }
+        return false;
+    }
```
HttpSession에 대해서 자세히 알아보시면 좋을 것 같습니다. HttpServletRequest에 의존할 필요가 없어보이네요 :)

Jeonminje replied:
넵 감사합니다:)

sunwootest commented on src/main/java/study/movieservice/controller/LoginController.java:
```diff
@@ -0,0 +1,36 @@
+
+        if(memberService.logIn(member.getLoginId(), member.getLoginPassword(), request)){
+            return "로그인이 완료되었습니다.";
+        }
+        return "아이디 또는 비밀번호가 일치하지 않습니다.";
```
아이디와 비밀번호가 일치하지 않은데 200 http 상태코드를 보내주네요 :)

Jeonminje replied:
미처 생각치 못한 것 같네요 수정하겠습니다

sunwootest commented on src/main/java/study/movieservice/controller/LoginController.java:
```diff
@@ -0,0 +1,36 @@
+
+    private final MemberService memberService;
+
+    @PostMapping("/log-in")
+    public String login(@RequestBody MemberDTO member, HttpServletRequest request){
```
로그인을 하기 위해선 아이디와 비밀번호만 필요한데 MemberDTO 전부가 필요한가요?

Jeonminje replied:
혹시 @RequestParam을 이용해 데이터를 받자니 보안 부분이 신경쓰이는데 로그인관련 DTO를 따로 만들어 제이슨 형식으로 받는게 좋을까요?

sunwootest replied:
네 그렇게 진행하는게 나아보입니다 :)

Jeonminje replied:
넵 감사합니다!

sunwootest commented on src/main/resources/study/movieservice/repository/MemberMapper.xml:
```diff
@@ -14,5 +14,11 @@
     </select>
+
+    <select id="getMember" resultType="Member">
+        select id, email, login_id, login_password, nickname, grade
+        from members
+        where login_id = #{loginId}
```
보통 테이블명은 복수형이 아닌 단수형으로 작성합니다.

Jeonminje replied:
넵 수정하겠습니다!

Commit:
Merge branch 'SignUp-emailCheck-minje' into SignUp-login-minje

Commit:
로그인 컨트롤러 및 로그인 서비스 관련 함수 수정

## LoginController
- HttpServletRequest에 의존하던 코드를 HttpSession을 직접 받아 사용하는 것으로 변경하였습니다.
- 로그인 실패시 200코드를 반환해주던 것에서 오류코드를 리턴해주도록 변경하였습니다.
- 아이디와 비밀번호를 받기위해 LoginParam 클래스를 추가해 DTO로 이용하였습니다.

## LoginParam
- 아이디와 비밀번호를 받기위한 DTO로 이번에 추가했습니다.

## MemberMapper.xml
- 테이블 명을 members -> member로 바꾸고 관련 쿼리를 수정하였습니다.

## SessionConst
- 세션의 키값을 가지고 있는 클래스를 만들어 주었습니다. (닉네임, 로그인아이디)

## MemberService
- logIn()에서 기존 세션에 key = 아이디, value = 닉네임 으로 저장하던 것에서 각각 value로 저장하도록 수정했습니다. 또한 기존 HttpServletRequest 관련 코드를 HttpSession으로 수정하였습니다.
- logOut() : 기존 loginId를 파라미터로 받아 로그인아이디가 key로 저장된 세션에서 삭제를 해주는 방식에서 HttpSession.invalidate() 함수를 이용해 로그아웃 시 세션을 비우는 방식으로 수정하였습니다.

Jeonminje commented:
## LoginController
- HttpServletRequest에 의존하던 코드를 HttpSession을 직접 받아 사용하는 것으로 변경하였습니다.
- 로그인 실패시 200코드를 반환해주던 것에서 오류코드를 리턴해주도록 변경하였습니다.
- 아이디와 비밀번호를 받기위해 LoginParam 클래스를 추가해 DTO로 이용하였습니다.

## LoginParam
- 아이디와 비밀번호를 받기위한 DTO로 이번에 추가했습니다.

## MemberMapper.xml
- 테이블 명을 members -> member로 바꾸고 관련 쿼리를 수정하였습니다.

## SessionConst
- 세션의 키값을 가지고 있는 클래스를 만들어 주었습니다. (닉네임, 로그인아이디)

## MemberService
- logIn()에서 기존 세션에 key = 아이디, value = 닉네임 으로 저장하던 것에서 각각 value로 저장하도록 수정했습니다. 또한 기존 HttpServletRequest 관련 코드를 HttpSession으로 수정하였습니다.
- logOut() : 기존 loginId를 파라미터로 받아 로그인아이디가 key로 저장된 세션에서 삭제를 해주는 방식에서 HttpSession.invalidate() 함수를 이용해 로그아웃 시 세션을 비우는 방식으로 수정하였습니다.

Jeonminje commented:
커밋했습니다!

sunwootest commented on src/main/java/study/movieservice/domain/SessionConst.java:
```diff
@@ -0,0 +1,9 @@
+
+public class SessionConst {
+
+    public static final String LOGIN_ID = "loginId";
+    public static final String NICKNAME = "Nickname";
```
하나는 카멜케이스이고 다른 하나는 그냥 대문자로 시작되네요. 세션 key값도 통일되면 좋겠습니다 :)

Jeonminje replied:
넵 감사합니다!

sunwootest commented on src/main/java/study/movieservice/repository/MemberMapper.java:
```diff
@@ -3,10 +3,12 @@
 public interface MemberMapper {
     void save(Member member);
     boolean findByLoginId(String loginId);
+    Optional<Member> getMember(String loginId);
```
getMember라는 메소드가 모호한 것 같습니다. PK값인 id를 통해 Member를 조회하는 것이 아닌 loginId로 조회하기 때문에 다른 방식으로 메소드명을 정하는 것이 나을 것 같습니다.

Jeonminje replied:
수정하도록 하겠습니다..!

sunwootest commented on src/main/java/study/movieservice/service/MemberService.java:
```diff
@@ -44,4 +48,27 @@ public void sendSignUpMessage(String email){
+                return true;
+            }
+        }
+        return false;
```
login이 실패하면 예외를 발생시켜서 바로 ResponseEntity를 전달하게 할 수도 있습니다. ExceptionHandler 부분을 약간만 수정하면 될거에요. 그렇게 되면 Controller의 코드도 간결해질겁니다.

Jeonminje replied:
넵 반영하도록 하겠습니다

sunwootest commented on src/main/java/study/movieservice/service/MemberService.java:
```diff
@@ -44,4 +48,27 @@ public void sendSignUpMessage(String email){
+            Member findMember = finding.get();
+            if(BCrypt.checkpw(password, findMember.getLoginPassword())){
+
+                session.setAttribute(SessionConst.LOGIN_ID, findMember.getLoginId());
+                session.setAttribute(SessionConst.NICKNAME, findMember.getNickname());
```
MemberService인데 HttpSession에 의존적인 코드가 되었네요. 세션과 관련된 부분은 다른 클래스로 분리해도 괜찮을 것 같습니다.

sunwootest commented on src/main/java/study/movieservice/service/MemberService.java:
```diff
@@ -44,4 +48,27 @@ public void sendSignUpMessage(String email){
+        return false;
+    }
+
+    public void logOut(HttpSession session){
+        if(session != null){
+            session.invalidate();
+        }
```
해당 클래스에서 HttpSession을 빈으로 주입받으면 굳이 logOut이라는 메소드이 파라미터로 HttpSession을 받을 필요는 없습니다. 물론 세션과 관련된 클래스를 분리하는 것이 가장 나을 것 같고요.

Jeonminje replied:
분리하도록 하겠습니다!

sunwootest commented on src/main/java/study/movieservice/domain/SessionConst.java:
```diff
@@ -0,0 +1,9 @@
+    public static final String LOGIN_ID = "loginId";
+    public static final String NICKNAME = "Nickname";
+
+    private SessionConst(){}
```
어떤 의도로 이렇게 만드셨나요?

Jeonminje replied:
로그인 이후 닉네임과 로그인 아이디가 리뷰 작성시 다른사람에게 노출되거나 추후에 사용 될 일이 있을 수도있을거 같아, 로그인아이디와 닉네임을 세션으로 저장하기 위한 키입니다. 키값에 의존하지 않게 하기위해 따로 세션에 키값으로 사용하기위한 상수를 저장하는 클래스를 만들었습니다. 
 상수를 위한 클래스 이기때문에 인스턴스가 생성되면 안된다고 판단해 private 생성자를 이용하였습니다.

sunwootest commented on src/main/java/study/movieservice/controller/LoginController.java:
```diff
@@ -0,0 +1,40 @@
+    }
+
+    @PostMapping("/log-out")
+    public String logout(HttpSession session){
+        memberService.logOut(session);
+        return "로그아웃 되었습니다.";
+    }
```
HttpSession은 빈으로 등록되어있습니다.

Jeonminje committed:
예외 추가

## exception 폴더를 만들어 예외 추가 ( 모두 런타임 예외 )
- LoginIncorrectException : 아이디나 비밀번호가 일치하지 않을 경우 발생하는 예외 입니다.
- LoginSetAttributeException : 세션의 key, value를 세팅할 때 예외가 발생하면 이 예외로 변환을 해줍니다.
- LogoutException : 로그아웃시 발생하는 예외 입니다.

## LoginController
- 로그인이 실패하면 예외를 발생시켜 ExceptionHandler에서 응답을 하도록 해주었습니다.

## MemberMapper
- getMember() 함수의 명을 수정하였습니다.

## MemberService
- logOut(), logIn() 함수에서 HttpSession 관련 로직을 분리하였습니다. SessionManager를 빈으로 주입받아 사용 합니다.

## SessionManager
- HttpSession을 빈으로 주입받아 세션관련 로직을 수행해줍니다.

Jeonminje commented:
## exception 폴더를 만들어 예외 추가 ( 모두 런타임 예외 )
- LoginIncorrectException : 아이디나 비밀번호가 일치하지 않을 경우 발생하는 예외 입니다.
- LoginSetAttributeException : 세션의 key, value를 세팅할 때 예외가 발생하면 이 예외로 변환을 해줍니다.
- LogoutException : 로그아웃시 발생하는 예외 입니다.

## LoginController
- 로그인이 실패하면 예외를 발생시켜 ExceptionHandler에서 응답을 하도록 해주었습니다.

## MemberMapper
- getMember() 함수의 명을 수정하였습니다.

## MemberService
- logOut(), logIn() 함수에서 HttpSession 관련 로직을 분리하였습니다. SessionManager를 빈으로 주입받아 사용 합니다.

## SessionManager
- HttpSession을 빈으로 주입받아 세션관련 로직을 수행해줍니다.

Jeonminje commented:
커밋했습니다!

sunwootest commented on src/main/java/study/movieservice/exception/LogoutException.java:
```diff
@@ -0,0 +1,6 @@
+package study.movieservice.exception;
+
+public class LogoutException extends RuntimeException{
+    public LogoutException() {
+    }
+}
```
매번 예외상황이 발생할 때마다 새로운 Exception을 만들 수는 없습니다. 최대한 java에서 기본적으로 제공되는 Exception을 사용하도록하고 가능하면 구체적인 예외를 상속하도록 하는 것이 좋습니다. Exception의 매개변수로 예외와 관련된 메시지와 상태코드를 전달하는 방법도 사용할 수 있겠네요. 이건 어디까지나 하나의 예시일 뿐 다른 좋은 방법이 있다면 고민하고 도입하시면됩니다.

Jeonminje replied:
말씀감사합니다!
혹시 Exception의 매개변수로 관련 메시지를 전달해 준다면, 예외를 던지는 코드가 예외 메시지에 의존한다 생각하여 핸들러에서 직접 작성해 주었는데 이러한 방식은 효율적이지 않은 방식일까요?

sunwootest replied:
LogoutException에 새로운 생성자를 만들어주고 정보를 담을 수 있는 필드를 만들어주면 해결할 수 있겠네요. 그리고 해당 정보는 상수처럼 다른 곳에 저장해두고요.

Jeonminje replied:
감사합니다!

sunwootest commented on src/main/java/study/movieservice/service/SessionManager.java:
```diff
@@ -0,0 +1,34 @@
+    public void storeLoginIdAndNickname(String loginId, String nickname){
+        try{
+            session.setAttribute(SessionConst.LOGIN_ID, loginId);
+            session.setAttribute(SessionConst.NICKNAME, nickname);
```
세션에 닉네임은 어떤 목적으로 저장하나요?

Jeonminje replied:
추후에 리뷰작성이나 마이페이지 등 닉네임을 활용하거나 노출 시킬 일이 있을거 같습니다. 그때 그때 DB에서 nickname을 가져오는 것보다 효율적이라 생각하여 세션에 저장하였습니다.

sunwootest commented on src/main/java/study/movieservice/service/SessionManager.java:
```diff
@@ -0,0 +1,34 @@
+    public void deleteAll(){
+        try{
+            session.invalidate();
+        }catch (Exception e){
+            throw new LogoutException();
+        }
```
어떤 목적으로 예외처리를 작성하셨나요?

Jeonminje replied:
세션 자체나 세션 객체에 저희가 알 수 없는 원인으로 작동하지 않을 경우 서버가 멈추는 것을 대비해 예외처리를 작성하였습니다!

sunwootest commented on src/main/java/study/movieservice/domain/SessionConst.java:
```diff
@@ -0,0 +1,9 @@
+public class SessionConst {
+
+    public static final String LOGIN_ID = "loginId";
+    public static final String NICKNAME = "nickname";
```
이름 컨벤션 리뷰반영이 안되었네요 :)

Jeonminje replied:
상수명은 대문자에 snakeCase를 적용하였고 값인 loginId 과 nickname 모두 camelcase를 적용하였습니다! 혹시 어떤 컨벤션을 말씀하시는 건지 여쭤봐도 될까요?

sunwootest replied:
제가 잘못봤네요 ㅎㅎ 이 부분은 그냥 진행하면 됩니다 :)

sunwootest commented on src/main/java/study/movieservice/controller/LoginController.java:
```diff
@@ -0,0 +1,31 @@
+    @PostMapping("/log-out")
+    public String logout(){
+        memberService.logOut();
+        return "로그아웃 되었습니다.";
```
로그인에서는 ResponseEntity로 전송하고 logout은 String만 전송하네요.

Jeonminje replied:
수정하겠습니다!

Jeonminje committed:
GeneralExceptionHandler 추가 및 예외 처리 수정
1. 기존 controller에서 ResponseEntity 를 리턴하던 것에서 String을 리턴하는 것으로 방식을 바꾸고 @ResponseStatus 어노테이션을 이용하여 상태를 응답해주었습니다.
2. controller에 있던 비즈니스 로직을 모두 제거하였습니다. ( 예외는 모두 service단에서 던져주었습니다.)
3. EexceptionMessageConst 이넘 클래스를 생성하여 예외 메시지를 분리해주었습니다.
4. 기존 RuntimeException을 상속하여 직접 만들어 주었던 예외들을 모두 삭제, 및 만들어져 있는 예외를 사용하는 방안으로 변환해주었습니다.
5. 로그인 시, 세션에 저장하는 값을 로그인아이디가 아닌 primary key 값인 memberId를 저장해주었습니다.

Jeonminje commented:
# 요약
1. 기존 controller에서 ResponseEntity 를 리턴하던 것에서 String을 리턴하는 것으로 방식을 바꾸고 @ResponseStatus 어노테이션을 이용하여 상태를 응답해주었습니다. 
2. controller에 있던 비즈니스 로직을 모두 제거하였습니다. ( 예외는 모두 service단에서 던져주었습니다.)
3. EexceptionMessageConst 이넘 클래스를 생성하여 예외 메시지를 분리해주었습니다.
4. 기존 RuntimeException을 상속하여 직접 만들어 주었던 예외들을 모두 삭제, 및 만들어져 있는 예외를 사용하는 방안으로 변환해주었습니다.
5. 로그인 시, 세션에 저장하는 값을 로그인아이디가 아닌 primary key 값인 memberId를 저장해주었습니다.

# 코드 변경
### controller.LoginController
```java
@PostMapping("/log-in")
    public ResponseEntity<String> login(@RequestBody LoginParam loginParam){
        memberService.logIn(loginParam.getLoginId(), loginParam.getLoginPassword());
        return ResponseEntity.ok("로그인이 완료되었습니다.");
    }

    @PostMapping("/log-out")
    public String logout(){
        memberService.logOut();
        return "로그아웃 되었습니다.";
    }
```
변경 후 ( 예외 처리로직을 없애주었습니다. )
```java
@PostMapping("/log-in")
    @ResponseStatus(HttpStatus.OK)
    public String login(@RequestBody LoginParam loginParam) throws LoginException {
        memberService.logIn(loginParam.getLoginId(), loginParam.getLoginPassword());
        return SUCCESS_LOGIN.getMessage();
    }

    @PostMapping("/log-out")
    @ResponseStatus(HttpStatus.OK)
    public String logout(){
        memberService.logOut();
        return SUCCESS_LOGOUT.getMessage();
    }
```

### controller.MemberController
``` java
@PostMapping("/id-check")
    public ResponseEntity<String> loginIdCheck(@RequestParam String loginId) throws LoginException {

        if (memberService.checkLoginId(loginId)) {
            throw new LoginException("이미 사용중인 아이디 입니다.");
        } else {
            return new ResponseEntity<>("사용가능한 아이디 입니다.", HttpStatus.OK);
        }
    }

@ResponseBody
    @PostMapping("/email-code")
    public ResponseEntity<String> memberEmailCodeCheck(@RequestParam String email, @RequestParam String emailCode) {
        boolean result = mailService.checkEmailCode(email, emailCode);

        if(result){
            return ResponseEntity.ok("인증이 완료되었습니다.");
        }else{
            return new ResponseEntity<>("인증번호가 틀렸습니다. 다시 시도해주세요", HttpStatus.BAD_REQUEST);
        }
    }
```
변경 후
``` java
    @PostMapping("/id-check")
    @ResponseStatus(HttpStatus.OK)
    public String loginIdCheck(@RequestParam String loginId) {
        memberService.checkLoginId(loginId);
        return SUCCESS_CHECK_ID.getMessage();
    }

    @PostMapping("/email-code")
    @ResponseStatus(HttpStatus.OK)
    public String memberEmailCodeCheck(@RequestParam String email, @RequestParam String emailCode) {
        mailService.checkEmailCode(email, emailCode);
        return SUCCESS_CHECK_EMAIL.getMessage();
    }
```

### domain.SessionConst
```java
 public static final String LOGIN_ID = "loginId";
```
변경 후 ( 로그인아이디 대신 프라이머리 키 값을 세션에 저장할 것입니다. )
```
    public static final String MEMBER_ID = "memberId";
```
### exception.GeneralExceptionHandler
```java
@ControllerAdvice
public class GeneralExceptionHandler {

    @ExceptionHandler(LoginException.class)
    public ResponseEntity<String> loginHandler(LoginException e) {
        return new ResponseEntity<>(e.getMessage(), HttpStatus.CONFLICT);
    }
}
```
변경 후 ( ResponseEntity를 사용하지 않고 어노테이션을 사용하여 상태코드를 처리하였습니다. )
```java
@RestControllerAdvice
public class GeneralExceptionHandler {
    @ExceptionHandler(LoginException.class)
    @ResponseStatus(HttpStatus.BAD_REQUEST)
    public String loginHandler(LoginException e) {
        return e.getMessage();
    }
}
```
### service.MailService
```java
public void sendMail(MimeMessage message) {
        try{
            emailSender.send(message);
        }catch (Exception e){
            throw new MailSendException("이메일 전송에 실패하였습니다.");
        }
    }

    public boolean checkEmailCode(String email, String emailCode){
        ValueOperations<String,String > store = redisTemplate.opsForValue();
        String code = store.get(email);

        if(code != null && code.equals(emailCode)){
            redisTemplate.delete(email);
            return true;
        }else{
            return false;
        }
    }
```
변경 후 ( enum 클래스를 이용해 예외메시지를 처리하고 컨트롤러에게 boolean 값을 넘기는 대신 예외를 직접 던져주었습니다. )
``` java
    public void sendMail(MimeMessage message) {
        try{
            emailSender.send(message);
        }catch (Exception e){
            throw new MailSendException(FAILED_SEND_EMAIL.getMessage());
        }
    }

    public void checkEmailCode(String email, String emailCode){
        ValueOperations<String,String > store = redisTemplate.opsForValue();
        String code = store.get(email);

        if(code != null && code.equals(emailCode)){
            redisTemplate.delete(email);
        }else{
            throw new MailAuthenticationException(FAILED_CHECK_EMAIL.getMessage());
        }
    }
```

### service.MemberService
```java
 public boolean checkLoginId(String loginId){
        boolean findFlag  = memberMapper.findByLoginId(loginId);

        if(findFlag){
            return true;
        }

        return false;
    }
```
변경 후 ( 컨트롤러에게 비즈니스 로직을 미루지 않고 직접 예외를 던져주었습니다. )
```java
    public void checkLoginId(String loginId) {
        boolean findFlag  = memberMapper.findByLoginId(loginId);

        if(findFlag){
            throw new IllegalArgumentException(FAILED_CHECK_ID.getMessage());
        }
    }
```

### service.SessionManager
```java
public void storeLoginIdAndNickname(String loginId, String nickname){
        try{
            session.setAttribute(SessionConst.LOGIN_ID, loginId);
            session.setAttribute(SessionConst.NICKNAME, nickname);
        } catch (Exception e){
            throw new LoginSetAttributeException();
        }

    }
```
변경 후 ( 불필요한 예외 제거 및 세션 저장시 로그인 아이디 대신 primary key를 저장 하는 것으로 변경하였습니다. )
```java
public void storeLoginIdAndNickname(Long memberId, String nickname){
        session.setAttribute(SessionConst.MEMBER_ID, memberId);
        session.setAttribute(SessionConst.NICKNAME, nickname);
    }
```

Jeonminje commented:
추가로 몇몇 메서드 정의 순서를 변경하였습니다!

nahokyun commented on src/main/java/study/movieservice/service/SessionManager.java:
```diff
@@ -0,0 +1,23 @@
+
+    public void storeLoginIdAndNickname(Long memberId, String nickname){
+        session.setAttribute(SessionConst.MEMBER_ID, memberId);
+        session.setAttribute(SessionConst.NICKNAME, nickname);
```
세션 또한 서버자원을 차지하고, 키에대한 정보 하나만으로 사용자 데이터를 가져올수 있기에 보안에 상대적으로 뛰어나다고 알고있는데 세션키로 닉네임 정보를 추가로 넣는것은 목적과 방향이 조금 다르다고 생각합니다!

Jeonminje replied:
고민해보니 닉네임을 항상 사용하는 값이 아닐뿐 더러 일리있는 말 같습니다 수정하도록 하겠습니다!

sunwootest replied:
이 부분은 충분히 고려할만한 요소네요 :)

sunwootest commented:
저는 임시 approve 드립니다. 팀원끼리 회의하셔서 수정할 부분들은 수정하세요 :)

Jeonminje commented:
감사합니다!

Jeonminje committed:
log-in시 session에 nickname과 memberId를 저장하던 것을 닉네임은 저장하지 않는 코드로 수정.

Jeonminje committed:
SessionConst의 nickname관련 상수 제거
